src/rpc/methods/eth.rs

```diff
@@ -3169,6 +3170,117 @@ impl RpcMethod<2> for EthTraceReplayBlockTransactions {
     }
 }
 
+fn get_eth_block_number_from_string<DB: Blockstore>(
+    chain_store: &ChainStore<DB>,
+    block: Option<&str>,
+) -> Result<EthUint64> {
+    let head = chain_store.heaviest_tipset();
+    let block_value = block.unwrap_or("latest");
+    match block_value {
+        "earliest" => bail!("block param \"earliest\" is not supported"),
+        "pending" => Ok(EthUint64(head.epoch() as u64)),
+        "latest" => {
+            let parent = chain_store
+                .chain_index
+                .load_required_tipset(head.parents())
+                .context("cannot get parent tipset")?;
+            Ok(EthUint64(parent.epoch() as u64))
+        }
+        "safe" => {
+            let latest_height = head.epoch() - 1;
+            let safe_height = latest_height - SAFE_EPOCH_DELAY;
+            Ok(EthUint64(safe_height as u64))
+        }
+        _ => {
+            let block_num = EthUint64(
+                hex_str_to_epoch(block_value).context("cannot parse block number")? as u64,
+            );
+            Ok(block_num)
+        }
+    }
+}
+
+pub enum EthTraceFilter {}
+impl RpcMethod<1> for EthTraceFilter {
+    const N_REQUIRED_PARAMS: usize = 1;
+    const NAME: &'static str = "Filecoin.EthTraceFilter";
+    const NAME_ALIAS: Option<&'static str> = Some("trace_filter");
+    const PARAM_NAMES: [&'static str; 1] = ["filter"];
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Read;
+    type Params = (EthTraceFilterCriteria,);
+    type Ok = Vec<EthBlockTrace>;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (filter,): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let from_block =
+            get_eth_block_number_from_string(ctx.chain_store(), filter.from_block.as_deref())
+                .context("cannot parse fromBlock")?;
+
+        let to_block =
+            get_eth_block_number_from_string(ctx.chain_store(), filter.to_block.as_deref())
+                .context("cannot parse toBlock")?;
+        Ok(trace_filter(ctx, filter, from_block, to_block).await?)
+    }
+}
+
+async fn trace_filter(
+    ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+    filter: EthTraceFilterCriteria,
+    from_block: EthUint64,
+    to_block: EthUint64,
+) -> Result<Vec<EthBlockTrace>> {
+    let mut results = vec![];
+    if let Some(EthUint64(0)) = filter.count {
+        return Ok(results);
+    }
+    let count = filter.count.clone().unwrap_or(EthUint64(0)).0;
+    let trace_filter_max_results = env_or_default("FOREST_TRACE_FILTER_MAX_RESULT", 500);
```

Comment:
Not great to parse env value on every call. Better to make `trace_filter_max_results` static

Reply:
fixed